### PR TITLE
[QATM-1974] Corregir el error que había con la comparación de versiones

### PR DESCRIPTION
**Pull request by mraposo-stratio**

### CODE REVIEW

File: src/main/java/com/stratio/qa/aspects/RunOnTagAspect.java [hunk 1]
```diff
@@ -144,251 +144,236 @@ public boolean checkParams(String[][] params) throws Exception {
...
-                            result = result && false;
-                        } else {
-                            result = result || false;
-                        }
+                }
+            } else {
+                String[] parver = System.getProperty(param).split("\\.");
+                String[] valver = value.split("\\.");
+                if (parver.length != valver.length) {
+                    throw new Exception("Error while parsing params. The versions must have the same numbers of elements");
```
Comment (aalfonso-stratio) [inline src/main/java/com/stratio/qa/aspects/RunOnTagAspect.java]: ... same number of elements

File: src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java [hunk 2]
```diff
@@ -48,17 +48,33 @@ public void testCheckParams_2() throws Exception {
...
     public void testGetEmptyParams() throws Exception {
         assertThatExceptionOfType(Exception.class).isThrownBy(() -> runontag.getParams("@runOnEnv"))
                 .withMessage("Error while parsing params. Format is: \"runOnEnv(PARAM)\", but found: " + "@runOnEnv");
     }
 
+    @Test
+    public void testCheckLengthVersion() throws Exception {
+        System.setProperty("VERSION","1.0.0");
+        assertThatExceptionOfType(Exception.class).isThrownBy(() -> runontag.checkParams(runontag.getParams("@runOnEnv(VERSION>1.0)")))
+                .withMessage("Error while parsing params. The versions must have the same numbers of elements");
```
Comment (aalfonso-stratio) [inline src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java]: ... the same number of elements

File: src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java [hunk 3]
```diff
@@ -48,17 +48,33 @@ public void testCheckParams_2() throws Exception {
...
+    }
+
+    @Test
+    public void testCheckLengthVersion_2() throws Exception {
+        System.setProperty("VERSION","1.0.0-1.0.0");
+        assertThatExceptionOfType(Exception.class).isThrownBy(() -> runontag.checkParams(runontag.getParams("@runOnEnv(VERSION>1.0.0-1.0)")))
+                .withMessage("Error while parsing params. The versions must have the same numbers of elements");
```
Comment (aalfonso-stratio) [inline src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java]: ... the same number of elements

File: src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java (hunk 2 above)
Comment (mraposo-stratio) [reply src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java]: Cambiado

File: src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java (hunk 3 above)
Comment (mraposo-stratio) [reply src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java]: Cambiado

File: src/main/java/com/stratio/qa/aspects/RunOnTagAspect.java [hunk 4]
```diff
@@ -144,251 +145,239 @@ public boolean checkParams(String[][] params) throws Exception {
...
+        } else {
+            return result || valor;
+        }
+    }
+
+    private boolean checkVersion(char operador, String param, String value) throws Exception {
+        boolean result = true;
+        String regexp = "^[[[0-9]+.]+[0-9]+][-[[0-9]+.]+[0-9]+]*";
+        if (!Pattern.matches(regexp, System.getProperty(param)) || !Pattern.matches(regexp, value)) {
+            throw new Exception("Error while parsing params. The versions has some caracter that it isn't number,.,-");
```
Comment (aalfonso-stratio) [inline src/main/java/com/stratio/qa/aspects/RunOnTagAspect.java]: Error while parsing params. The versions have some characters that are not numbers, '.' or '-'

File: src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java [hunk 5]
```diff
@@ -48,17 +48,61 @@ public void testCheckParams_2() throws Exception {
...
+        System.setProperty("VERSION","1.0.0@1.0.0");
+        assertThatExceptionOfType(Exception.class).isThrownBy(() -> runontag.checkParams(runontag.getParams("@runOnEnv(VERSION>1.0.0-1.0)")))
+                .withMessage("Error while parsing params. The versions has some caracter that it isn't number,.,-");
+    }
+
+    @Test
+    public void testCheckVersion_3() throws Exception {
+        System.setProperty("VERSION","1.0.0-1.0.0");
+        assertThatExceptionOfType(Exception.class).isThrownBy(() -> runontag.checkParams(runontag.getParams("@runOnEnv(VERSION>1.0.*)")))
+                .withMessage("Error while parsing params. The versions has some caracter that it isn't number,.,-");
```
Comment (aalfonso-stratio) [inline src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java]: same as above

File: src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java [hunk 6]
```diff
@@ -48,17 +48,61 @@ public void testCheckParams_2() throws Exception {
...
+    }
+
+    @Test
+    public void testCheckVersion_4() throws Exception {
+        System.setProperty("VERSION","1.0.0-1.0.0");
+        assertThatExceptionOfType(Exception.class).isThrownBy(() -> runontag.checkParams(runontag.getParams("@runOnEnv(VERSION>1.0.0-0.5.15-/+-)")))
+                .withMessage("Error while parsing params. The versions has some caracter that it isn't number,.,-");
```
Comment (aalfonso-stratio) [inline src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java]: same as above

File: src/main/java/com/stratio/qa/aspects/RunOnTagAspect.java (hunk 4 above)
Comment (mraposo-stratio) [reply src/main/java/com/stratio/qa/aspects/RunOnTagAspect.java]: Resuelto

File: src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java (hunk 5 above)
Comment (mraposo-stratio) [reply src/test/java/com/stratio/qa/aspects/RunOnEnvTagAspectTest.java]: Resuelto

**Pull request merged**